server/api/cart.js

```diff
@@ -108,6 +108,131 @@ router.post('/', async (req, res, next) => {
   }
 })
 
+/*
+POST: /api/cart/add
+adds the product of a specified quantity to the cart.
+
+example req.body is:
+{
+  productId : 3,
+  quantity : 1,
+}
+*/
+
+router.post('/add', async (req, res, next) => {
+//check if product exists
+  const productData = await Product.findById(req.body.productId)
+  if (!productData) {
+    throw new Error('product of specified id does not exist')
+  }
+
+  //if not a user
+  if (!req.user) {
+
+  //find index of the cart in our session
+    const itemIdx = req.session.cart.findIndex((item) => {
+      return item.productId === req.body.productId
+    })
+
+    //update or create cartItemObject in our session array
+    if (itemIdx !== -1) {
+    //if the instance of the product exists:
+      //adds the quantity to the existing quantity
+      req.session.cart[itemIdx].quantity += req.body.quantity
+    } else {
+    //push the new cart object into the session
+      req.session.cart.push({
+        productId: req.body.productId,
+        quantity: req.body.quantity,
+        product: productData,
+      })
+    }
+
+    res.status(201).json(req.session.cart)
+
+  } else {
+  //if this is a user
+
+    //get the cartItemData, to get its id for use as the unique key
+    const cartItemData = await CartItem.findOne({
+      where: {
+        userId: req.user.id,
+        productId: req.body.productId
+      }
+    })
+
+    //assign id if found or undefined if notFound
+    const cartItemId = cartItemData ?
+      cartItemData.dataValues.id :
+      undefined
+
+    const cartItemQuantity = cartItemData ?
+      cartItemData.dataValues.quantity + req.body.quantity :
+      req.body.quantity
+
+    //upserts data into database by adding the quantity
+    await CartItem.upsert(
+      {
+        id: cartItemId,
+        userId: req.user.id,
+        productId: req.body.productId,
+        quantity: cartItemQuantity,
+      }
+    )
+
+    const cartItems = await getCart(req.user.id)
+    req.session.cart = cartItems
+    res.status(201).json(cartItems)
```

Comment:
move this line down to be between lines 186 and 187.
should read: 
            res.status(201).json(req.session.cart)